### ISSUE TITLE
Custom LearningRateSchedule not called within a Mirrored Strategy

### ISSUE THREAD

**Issue by houseofai** [tensorflow/tensorflow#44560]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): 
**From the tutorial [Custom training with tf.distribute.Strategy](https://www.tensorflow.org/tutorials/distribute/custom_training) with a Custom LearningRateSchedule**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Colab**
- TensorFlow version (use command below): **2.3.0**
- Python version: **3.6.9**
- GPU model and memory: **GPU from Colab**

**Describe the current behavior**
In a distributed environment, when adding a custom LearningRateSchedule to Adam optimizer to decay the learning rate over epochs, the learning rate is not decayed as expected.
I tested the same code in an environment without the distributed scope and the LR decay over the epochs.

**Describe the expected behavior**
The optimizer to call the LearningRateSchedule __call__ method to decay the learning rate after applying the gradients

**Standalone code to reproduce the issue**
Please check this Colab from the TF tutorial: https://colab.research.google.com/drive/1sT3MZ5hdGvsPhFQDrCFXzMR760HPCSsc?usp=sharing

See chapter: **Training loop**

**Other info / logs** 
```
class CustomSchedule(tf.keras.optimizers.schedules.LearningRateSchedule):
    def __init__(self):
      super(CustomSchedule, self).__init__()
      self.lr = 0.01

    def __call__(self, step):
      self.lr = self.lr/10
      print("NEW LR:", self.lr)
      return self.lr

with strategy.scope():
    model = create_model()
    optimizer = tf.keras.optimizers.Adam(learning_rate=CustomSchedule())
    checkpoint = tf.train.Checkpoint(optimizer=optimizer, model=model)
```
The `learning_rate=CustomSchedule()` is the only line I changed from the tutorial.

See the Colab for the **train_step**

**Comment by Saduf2019** [comment]
@houseofai 
Please share simple stand alone code to replicate the issue or if possible share a colab gist with the error reported.

**Comment by houseofai** [comment]
After a deeper analysis, I was able to see that the scheduler was called
